### PR TITLE
feat!: Use getLocale

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- uses `getLocale` from https://github.com/guardian/libs instead of requiring it to be passed to `init` 

## Why?

- cleaner API
- its the same technique host apps would use, but just handled internally

### CODE REVIEW

**Review by mxdvl — APPROVED** [review]
This is so much cleaner, we handle all locales: 🇬🇧 🇺🇸 🇦🇺 🇪🇺 !

**Review by ioanna0 — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -37,42 +34,34 @@ const init: InitCMP = ({
...
+			case 'tcfv2':
+			default:
+				// default is also 'tcfv2'
+				CMP = TCFv2;
+				break;
+		}
+
+		setCurrentFramework(framework);
+
+		CMP.init(pubData ?? {});
```
Comment (mxdvl) [inline src/index.ts:60]: Is it even worth passing an empty object?

```suggestion
		CMP.init(pubData);
```

**Review by mxdvl — COMMENTED** [review]
Simple and elegant.

I think there should we should decide on what kind of error we throw if old parameters are passed. Technically, we could simply say “`isInUsa` and `country` are ignored, assumed locale is `GB`, defined from @guardian/libs”.

There should also be tests that capture this behaviour. I know that some projects, such as `support-frontend`, which are relying on a different approach to the `@guardian/libs` to set the locale, might find it difficult to work with our new interface.

File: src/types/index.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export type InitCMP = (arg0: {
 	pubData?: PubData;
 	country?: Country;
```
Comment (mxdvl) [inline src/types/index.ts:10]:
```suggestion
```

**Review by mxdvl — COMMENTED** [review]
We also need to ensure that we mock the result of `getLocale` for the test page.

It might be as simple as setting the `gu.geolocation` value in the Svelte App.

**Comment by mxdvl** [comment]
Yea, it might be an improvement, but not urgent in any way an no consumer has requested this change so far.

**Pull request closed without merging**